x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	// XXX delete or uncomment this block before merge
+	//ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+)
+
+// Simple model, donate tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DonateToProviderValidatorSet(ctx sdk.Context) error {
+	if !k.shouldTransmit(ctx) {
+		return nil
+	}
+
+	port := k.GetDistributionTransmissionPort(ctx)
+	channel := k.GetDistributionTransmissionChannel(ctx)
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	providerFeePoolAddrStr := k.GetProviderFeePoolAddrStr(ctx)
+
+	tokens := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+	for _, token := range tokens {
+
+		// XXX delete or uncomment this block before merge
+		//if !strings.HasPrefix(token.Denom, "ibc/") {
+		//    denomTrace := ibctypes.ParseDenomTrace(token.Denom)
+		//    token.Denom = denomTrace.IBCDenom()
+		//}
+
+		err := k.ibcTransferKeeper.SendTransfer(ctx,
+			port,
+			channel,
+			token,
+			consumerFeePoolAddr,
+			providerFeePoolAddrStr,
+			clienttypes.Height{0, 0}, // XXX timeoutHeight
+			0,                        // XXX timeoutTimestamp
```

Comment:
@AdityaSripal is it appropriate to set these to 0 here? Also can I set the height as I have?

Reply:
No, don't set the height you have since timeout height is the height of the counterparty.

Use `ctx.BlockTime()` plus a relatively large timeoutDelta to set the absolute timeoutTimestamp for the packet

Reply:
in a normal transfer, it would be good UX for a transaction that hasn't gotten relayed to revert. A user may want to try something else in the face of a timed out transaction. in our scenario, what happens if it times out? ideally a relayer submits the timeout information and the tokens are put back into the consumerfeepooladdr. from there they could be included in a subsequent relayed transaction. is this any better than continuing to wait for the original unrelayed transaction? i'd say yes, because i make the assumption that the original relayed transaction was missed by a relayer but a subsequent attempt might not be—implying that it is easier for a relayer to do the basic action of relaying a new message, rather than recovering a lost message (unsubstantiated). if that is in fact the case then we want unrelayed messages to timeout (this may all be moot of course because the timeout needs to also be submitted by a relayer, right? or do timeouts undo themselves automatically?)

Reply:
I've changed this code to identical to the link @AdityaSripal provided, so the delta is 4 weeks. Seems a bit long I was going to make it an hour.